File: canteen-service/routes/github.ts
import { Request, Response } from "express";
import axios from "axios";

export async function getGitHubUser(req: Request, res: Response) {
  const user = req.query["username"];
  const data = await axios.get(`https://api.github.com/users/${user}`);

  if (data.data) {
    res.json(data.data);
  } else {
    res.status(404).json({
      code: "empty-github-user",
      id: "",
      message: "not found",
    });
  }
}

File: factory-service/routes/github.go
package routes

import (
	"fmt"
	"io"
	"net/http"

	"github.com/gin-gonic/gin"
)

func GetGitHubUser(c *gin.Context) {
	name := c.Query("username")

	fmt.Printf("query name is %s\n", name)
	url := fmt.Sprintf("https://api.github.com/users/%s", name)
	res, err := http.Get(url)
	if err != nil {
		print(err)
		c.JSON(200, gin.H{
			"message": "not found github user",
		})

		return
	}

	defer res.Body.Close()

	body, err := io.ReadAll(res.Body)
	if err != nil {
		print(err)
		c.JSON(200, gin.H{
			"message": "not found github user",
		})

		return
	}

	c.Writer.Header().Set("content-type", "application/json")
	c.Writer.Write(body)
}
